db/model/generator.js

```diff
@@ -484,12 +519,24 @@ module.exports = function generator(seq, dataTypes) {
    */
   Generator.prototype.assignToCollector = function () {
     const possibleCollectors = this.possibleCollectors;
+    let newColl = null;
     if (this.isActive && possibleCollectors && possibleCollectors.length) {
       possibleCollectors.sort((c1, c2) => c1.name > c2.name);
-      const newColl = possibleCollectors.find((c) => c.isRunning() && c.isAlive());
-      this.currentCollector = newColl ? newColl.name : null;
-    } else {
-      this.currentCollector = null;
+      newColl = possibleCollectors.find((c) => c.isRunning() && c.isAlive()) || null;
     }
+
+    // since currentCollector is an association we need to set the collectorId
+    // for this generator object.
+    this.collectorId = newColl ? newColl.id : null;
```

Comment:
I think setCollector would work, but it would return a promise and save to the db, which we don't want to do because it would require us to save to the db multiple times in a row and make the hook logic more complicated.

Can you update the comment to make it clear that we're setting the id directly to avoid saving to the db right now? And I think the comments for these two lines can be combined.

Something like: we could do setCurrentCollector, but we don't want to save to the db here, so instead we set the id directly and attach the collector object.